301-parameterized-linked-templates/detail/agtemplate.md

```diff
@@ -0,0 +1,100 @@
+# Use Linked Template to Limit Main Template Complexity (App Gateway)
+
+The template [paramlb.json](../nested/paramlb.json) is used to limit the complexity of the main template by encapsulating the deployment of an Application Gateway and only exposing parameters for the things that need to be changed for each deployment.  The template is only referenced one time in [azuredeploy.json](../azuredeploy.json).
+
+The values that need to change for each deployment are listed as parameters:
+```
+"parameters": {
+  "location": {
+    "type": "string",
+    "metadata": {
+      "description": "Azure region for App Gateway"
+    }
+  },
+  "lb_name": {
+    "type": "String"
+  },
+  "public_ip": {
+    "defaultValue": ""
+  },
+  "vnet_name": {
+    "type": "String"
+  },
+  "vnet_resource_group": {
+    "type": "String"
+  },
+  "vnet_subnet_name": {
+    "type": "String"
+  }
+},
+```
+
+The purpose of the Application Gateway in this template is to load balance incoming connections on port 80 (http) to all of the back-end nodes.  You can see the `frontendPorts` defined as a single listener on port 80:
+
+```
+"frontendPorts": [
+  {
+    "name": "port_80",
+    "properties": {
+      "port": 80
+    }
+  }
+],
+```
+```
+"httpListeners": [
+  {
+    "name": "http-listener",
+    "properties": {
+      "frontendIPConfiguration": {
+        "id": "[concat(resourceId('Microsoft.Network/applicationGateways', parameters('lb_name')), '/frontendIPConfigurations/appGwPublicFrontendIp')]"
```

Comment:
it might be best to remove code from here where you can, it's going to create a maint issue and you need to make sure it adhere's to the same best practices as the template - e.g. this needs to use resourceId not concat (as well as the snippets below)

Reply:
Removed all code snippets from the markdown files and replaced with references to the templates including line numbers.  Still a bit of a maintenance headache if line numbers change, but much less to keep up with compared to keeping the snippets up to date.